webapp/src/container.tsx

```diff
@@ -295,6 +302,7 @@ export class SettingsMenu extends data.Component<SettingsMenuProps, SettingsMenu
             || this.state.accessibleBlocks != nextState.accessibleBlocks
             || this.state.showShare != nextState.showShare
             || nextProps.inBlocks !== this.props.inBlocks
+            || !!nextState.extDownloadMenuItems?.length
```

Comment:
should this be checking it against whatever is in the current state instead of always updating?

Reply:
will revisit. the intention is for it to get latest menu items each time the menu opens, because they are contextual.

Reply:
Now grabs latest menu items when rendering, but for the items to refresh, I had to make it call forceUpdate when the dropdown opens.